syntax = "proto3";
package blob;
service Blob {
  rpc Init (InitRequest) returns (InitResponse) {}
  rpc Move (MoveRequest) returns (MoveResponse) {}
  rpc Region (RegionRequest) returns (RegionResponse) {}
}
message InitRequest {}
message InitResponse {
  string id = 1;
  double x = 2;
  double y = 3;
  int32 mass = 4;
}
message MoveRequest {
  string id = 1;
  double x = 2;
  double y = 3;
}
message MoveResponse {
  double x = 1;
  double y = 2;
  bool alive = 3;
  int32 mass = 4;
}
message RegionRequest {
  string id = 1;
  double x = 2;
  double y = 3;
}
message RegionResponse {
  repeated Player players = 1;
  repeated Food foods = 2;
}
message Player {
  string id = 1;
  double x = 2;
  double y = 3;
  bool alive = 4;
  int32 mass = 5;
}
message Food {
  double x = 1;
  double y = 2;
}
